jscomp/runtime/caml_splice_call.ml

```diff
@@ -38,6 +38,20 @@ let spliceApply : obj -> obj -> obj = [%raw{|function(fn,args){
   return fn.apply(null,applied)
 }|}] 
 
+let spliceNewApply : obj -> obj -> obj = [%raw{|function (ctor,args){
+  var i, argLen;
+  argLen = args.length
+  var applied = []
+  for(i = 0; i < argLen - 1; ++i){
+    applied.push(args[i])
+  }
+  var lastOne = args[argLen - 1]
+  for(i = 0; i < lastOne.length; ++i ){
+    applied.push(lastOne[i])
+  }
+  return new ctor(...applied)
```

Comment:
Since the spread operator (...) is the ES6 feature I think it's not suitable here. Maybe something like this should be here?

```js
function fn (ctor, args) {
  var i, argLen;
  argLen = args.length
  var applied = [null]
  for(i = 0; i < argLen - 1; ++i){
    applied.push(args[i])
  }
  var lastOne = args[argLen - 1]
  for(i = 0; i < lastOne.length; ++i){
    applied.push(lastOne[i])
  }
  var F = Function.prototype.bind.apply(ctor, applied)
  return new F()
}
```

Reply:
makes sense!